Language/Haskell/GhcMod/Monad.hs

```diff
@@ -52,17 +52,17 @@ import Exception
 import System.Directory
 import Prelude
 
-withGhcModEnv :: (IOish m, GmOut m) => FilePath -> Options -> (GhcModEnv -> m a) -> m a
+withGhcModEnv :: (IOish m, GmOut m) => FilePath -> Options -> ((GhcModEnv, GhcModLog)  -> m a) -> m a
 withGhcModEnv = withGhcModEnv' withCradle
  where
    withCradle dir =
-       gbracket (findCradle' dir) (liftIO . cleanupCradle)
-
-withGhcModEnv' :: (IOish m, GmOut m) => (FilePath -> (Cradle -> m a) -> m a) -> FilePath -> Options -> (GhcModEnv -> m a) -> m a
+       gbracket (runJournalT $ findCradle' dir) (liftIO . cleanupCradle . fst)
```

Comment:
Sorry I was not clear enough.

as it is defined, 

```
GhcModLog ml _ _ `mappend` GhcModLog ml' _ _ =
        GhcModLog (ml' `mplus` ml)
```

So given `gmLogLevel  (a:: GhcModLog) = Just la` and `gmLogLevel  (b:: GhcModLog) = Just lb`

```gmLogLevel  (a:: GhcModLog) <> (b::GhcModLog) =  Just lb```

the log level changes as we append GhcModLog which I thought was quite surprising.. why would the global log level change during execution ?